Sources/FoundationNetworking/URLSession/URLSession.swift

```diff
@@ -510,6 +511,20 @@ open class URLSession : NSObject {
      */
     @available(*, unavailable, message: "URLSessionStreamTask is not available in swift-corelibs-foundation")
     open func streamTask(withHostName hostname: String, port: Int) -> URLSessionStreamTask { NSUnsupported() }
+    
+    open func webSocketTask(with url: URL) -> URLSessionWebSocketTask {
+        return webSocketTask(with: _Request(url), behavior: .callDelegate)
+    }
+    
+    open func webSocketTask(with url: URL, protocols: [String]) -> URLSessionWebSocketTask {
+        var request = URLRequest(url: url)
+        request.setValue(protocols.joined(separator: ", "), forHTTPHeaderField: "Sec-WebSocket-Protocol")
```

Comment:
Are `Sec-WebSocket-Key` and `Sec-WebSocket-Version` added by curl?

Reply:
Correct.  -Key and -Version are properties of the WebSocket protocol itself.  -Protocol is meant for application-specific logic on top of the WebSocket protocol, which is why this belongs at our layer.